common/timer_migrate.go

```diff
@@ -35,3 +36,25 @@ func (tstore *TimerStore) MigrateVersion(ctx sdk.Context) (err error) {
 	tstore.setVersion(ctx, to)
 	return nil
 }
+
+// timerMigrate1to2: fix refcounts
+//   - convert entry keys from "<block>" to "<block>index (otherwise timer for an entry
+//     with some block will be overwritten by timer for another entry with same block.
+func timerMigrate1to2(ctx sdk.Context, tstore *TimerStore) error {
+	for _, which := range []types.TimerType{types.BlockHeight, types.BlockTime} {
+		store := tstore.getStoreTimer(ctx, which)
+
+		iterator := sdk.KVStorePrefixIterator(store, []byte{})
+		defer iterator.Close()
+
+		for ; iterator.Valid(); iterator.Next() {
+			value, key := types.DecodeBlockAndKey(iterator.Key())
+			key_v1 := types.EncodeKey(value)
+			key_v2 := types.EncodeBlockAndKey(value, key)
+			store.Set(key_v2, iterator.Value())
+			store.Delete(key_v1)
```

Comment:
why do we need key1? key and key1 are the same

Reply:
`key` and `key_v1` are not the same.  instead, `key_v1` and `iterator.Key()` are the same.
the assignment in line 52 can use `iterator.Key()` instead of re-encoding; but I think the latter is clearer to show the contrast with `key_v2`. I'll add a comment.